holoviews/ipython/display_hooks.py

```diff
@@ -146,7 +143,13 @@ def display_figure(fig, message=None, max_width='100%'):
     dpi = OutputMagic.options['dpi']
     backend = OutputMagic.options['backend']
 
-    if backend == 'd3' and mpld3:
+    if backend == 'nbagg':
+        import numpy as np
+        from matplotlib.backends.backend_nbagg import FigureManagerNbAgg, new_figure_manager_given_figure
+        manager = new_figure_manager_given_figure(np.random.randint(10**8), fig)
```

Comment:
You could create a numpy ``RandomState`` instance here with a fixed seed to get deterministic figure ids. Or use a counter...Or maybe keep track of the figure ids already used to avoid clashes.